### PR TITLE
Group fixes

### PR DESCRIPTION
**Pull request by aptkingston**
## Description
Opening for visibility as I'm shelving work on this for now. There are a few major refactors which still need done. This branch is functional but should not be considered complete.

The rest of the work which needs done is documented in #7043, as are all the tasks that have been completed so far and implemented on this branch.

### CODE REVIEW

**Comment by mike12345567** [comment]
recheck

File: packages/server/pm2.config.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 module.exports = {
   apps: [
     {
-      script: "dist/index.js",
+      script: "dist/index.ts",
```
Comment (Rory-Powell) [inline packages/server/pm2.config.js:4]: Is this right?

Comment (mike12345567) [reply packages/server/pm2.config.js:4]: No it isn't thanks for catching that, was an errant re-name!

**Review by Rory-Powell — APPROVED** [review]

**Comment by Rory-Powell** [comment]
Only some very minor comments which are mostly optional. Loving all the TS conversions as well 💙
Great job @aptkingston / @mike12345567!

**Review by shogunpurple — APPROVED** [review]
Great job on this - the new views and endpoints have taken a ton of work out of the client. I pulled this down with the `pro` changes etc and had a play around - everything is looking good! Looking forward to testing this one E2E on `release`.

**Pull request merged**